### ISSUE TITLE
NullReferenceException with Hangfire.Tags when scheduling a background job

### ISSUE THREAD

**Issue by Hulkstance** [face-it/Hangfire.Tags#82]
I'm encountering a `NullReferenceException` when trying to schedule a background job using Hangfire with Redis storage and Hangfire.Tags. The exception occurs when I attempt to schedule a job with `IBackgroundJobClient.Schedule` method.

Here's the exception stack trace:

```
Hangfire.BackgroundJobClientException: Background job creation failed. See inner exception for details.
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at Hangfire.Tags.States.CreateJobFilter.OnCreated(CreatedContext filterContext)
   at Hangfire.Client.BackgroundJobFactory.InvokeOnCreated(Tuple`2 x)
   at Hangfire.Profiling.ProfilerExtensions.InvokeAction[TInstance](InstanceAction`1 tuple)
   at Hangfire.Profiling.EmptyProfiler.InvokeMeasured[TInstance,TResult](TInstance instance, Func`2 action, String message)
   at Hangfire.Profiling.ProfilerExtensions.InvokeMeasured[TInstance](IProfiler profiler, TInstance instance, Action`1 action, String message)
   at Hangfire.Client.BackgroundJobFactory.InvokeClientFilter(IClientFilter filter, CreatingContext preContext, Func`1 continuation)
   at Hangfire.Client.BackgroundJobFactory.<>c__DisplayClass12_1.<CreateWithFilters>b__2()
   at Hangfire.Client.BackgroundJobFactory.InvokeClientFilter(IClientFilter filter, CreatingContext preContext, Func`1 continuation)
   at Hangfire.Client.BackgroundJobFactory.<>c__DisplayClass12_1.<CreateWithFilters>b__2()
   at Hangfire.Client.BackgroundJobFactory.CreateWithFilters(CreateContext context, IEnumerable`1 filters)
   at Hangfire.Client.BackgroundJobFactory.Create(CreateContext context)
   at Hangfire.BackgroundJobClient.Create(Job job, IState state, IDictionary`2 parameters)
   --- End of inner exception stack trace ---
   at Hangfire.BackgroundJobClient.Create(Job job, IState state, IDictionary`2 parameters)
   at Hangfire.BackgroundJobClient.Create(Job job, IState state)
   at Hangfire.BackgroundJobClientExtensions.Create[T](IBackgroundJobClient client, Expression`1 methodCall, IState state)
   at Hangfire.BackgroundJobClientExtensions.Schedule[T](IBackgroundJobClient client, Expression`1 methodCall, DateTimeOffset enqueueAt)
   at path\RecurringJobsInitializer.cs:line 26
```

I am scheduling the job like this:

```cs
public class RecurringJobsInitializer : BackgroundService
{
	readonly IRecurringJobManager _recurringJobManager;
	readonly IBackgroundJobClient _backgroundJobClient;
	
	public RecurringJobsInitializer(IRecurringJobManager recurringJobManager, IBackgroundJobClient backgroundJobClient)
	{
		_recurringJobManager = recurringJobManager;
		_backgroundJobClient = backgroundJobClient;
	}

	protected override Task ExecuteAsync(CancellationToken stoppingToken)
	{
		try
		{
			var jobTime = DateTime.UtcNow.AddMinutes(5);
			_backgroundJobClient.Schedule<DailyTradingHaltScheduler>(x => x.Perform(), jobTime);
		}
		catch (Exception ex)
		{
			throw; // int3 here
		}
		
		return Task.CompletedTask;
	}
}
```
And here's the `DailyTradingHaltScheduler` class definition:
```cs
[Queue("critical")]
[AutomaticRetry(Attempts = 3)]
public class DailyTradingHaltScheduler
{
	readonly IServiceProvider _serviceProvider;

	public DailyTradingHaltScheduler(IServiceProvider serviceProvider) => _serviceProvider = serviceProvider;

	public async Task Perform()
	{
                ...
        }
}
```

The versions of the NuGet packages I am using are:
```
<PackageReference Include="FaceIT.Hangfire.Tags" Version="1.8.5" />
<PackageReference Include="FaceIT.Hangfire.Tags.Pro.Redis" Version="1.9.1" />

<PackageReference Include="Hangfire.AspNetCore" Version="1.8.6" />
<PackageReference Include="Hangfire.Core" Version="1.8.6" />
<PackageReference Include="Hangfire.Pro" Version="3.0.0" />
<PackageReference Include="Hangfire.Pro.Redis" Version="3.0.3" />
<PackageReference Include="Hangfire.Throttling" Version="1.4.0" />
```

I've also found a similar issue here: https://github.com/face-it/Hangfire.Tags/issues/1

Here is the place where the problem occurs:
```cs
namespace Hangfire.Tags.Dashboard
{
    internal static class StorageRegistration
    {
        private static readonly Dictionary<JobStorage, Tuple<TagsOptions, ITagsServiceStorage>> ServiceStorages =
            new Dictionary<JobStorage, Tuple<TagsOptions, ITagsServiceStorage>>();

        internal static void Register(JobStorage jobStorage, TagsOptions options, ITagsServiceStorage serviceStorage)
        {
            ServiceStorages[jobStorage ?? JobStorage.Current] = new Tuple<TagsOptions, ITagsServiceStorage>(options, serviceStorage);
        }

        internal static Tuple<TagsOptions, ITagsServiceStorage> FindRegistration(this JobStorage storage)
        {
            if (ServiceStorages.TryGetValue(storage, out var jobStorageOptions))
                return jobStorageOptions;

            return ServiceStorages.TryGetValue(JobStorage.Current, out var currentOptions) ? currentOptions : default;
        }

    }
}
```

It's unable to match `storage` with the record from `ServiceStorages`.

![image](https://github.com/face-it/Hangfire.Tags/assets/33224919/d783f784-0f9a-4504-bc5d-294aa74feaf4)

![image](https://github.com/face-it/Hangfire.Tags/assets/33224919/679ee8f9-9f34-4368-a320-7b5a969cea1f)

I have the following configuration for Hangfire:

```cs
services.AddApplicationInsightsTelemetryForHangfire();

services.AddHangfire((serviceProvider, globalConfiguration) => globalConfiguration
	.SetDataCompatibilityLevel(CompatibilityLevel.Version_170)
	.UseRecommendedSerializerSettings()
	.UseRedisStorage(redisConnectionString, redisStorageOptions)
	.UseBatches()
	.UseThrottling(ThrottlingAction.RetryJob, 1.Seconds())
	.UseTagsWithRedis(new() { TagsListStyle = TagsListStyle.Dropdown }, redisStorageOptions)
	.UseApplicationInsightsTelemetry(serviceProvider));

services.AddHangfireServer((provider, options) =>
{
	options.Activator = new HangfireJobActivator(provider);
	options.WorkerCount = Environment.ProcessorCount * 5;
	options.HeartbeatInterval = 10.Seconds();
	options.SchedulePollingInterval = 1.Seconds();
	options.Queues = new[] { "critical", "default", "notifications" };
	options.ServerName = Environment.MachineName;
});
```

**Comment by nadyai** [comment]
It is fixed in latest prerelase 1.9.0-beta.3

I have the same exception for SQL server tags, when I use current latest stable:
<PackageReference Include="FaceIT.Hangfire.Tags.SqlServer" Version="1.8.2" />
and seems it is fixed in latest prerelase 1.9.0-beta.3. When I can expect a stable release with this fix?

The exception with Version="1.8.2" FaceIT.Hangfire.Tags.SqlServer
```
NullReferenceException: Object reference not set to an instance of an object.
Hangfire.Tags.States.CreateJobFilter.OnCreated(CreatedContext filterContext)
Hangfire.Client.BackgroundJobFactory.InvokeOnCreated(Tuple<IClientFilter, CreatedContext> x) in BackgroundJobFactory.cs
Hangfire.Profiling.ProfilerExtensions.InvokeAction<TInstance>(InstanceAction<TInstance> tuple) in ProfilerExtensions.cs
Hangfire.Profiling.ProfilerExtensions.InvokeMeasured<TInstance>(IProfiler profiler, TInstance instance, Action<TInstance> action, string message) in ProfilerExtensions.cs
Hangfire.Client.BackgroundJobFactory.InvokeClientFilter(IClientFilter filter, CreatingContext preContext, Func<CreatedContext> continuation) in BackgroundJobFactory.cs
Hangfire.Client.BackgroundJobFactory.InvokeClientFilter(IClientFilter filter, CreatingContext preContext, Func<CreatedContext> continuation) in BackgroundJobFactory.cs
Hangfire.Client.BackgroundJobFactory.Create(CreateContext context) in BackgroundJobFactory.cs
Hangfire.BackgroundJobClient.Create(Job job, IState state, IDictionary<string, object> parameters) in BackgroundJobClient.cs
```

**Comment by yongliu-mdsol** [comment]
we also encounter this issue. Would be nice to have a stable release for 1.9.0.
@r-win are there more things left for the branch hangfire_182 to be promoted?

**Comment by r-win** [comment]
Merged to 1.9.0
